plugins/inputs/nginx_plus/README.md

```diff
@@ -0,0 +1,125 @@
+# Telegraf Plugin: nginx_plus
+
+Nginx Plus is a commercial version of the open source web server Nginx. The use this plugin you will need a license. For more information about the differences between Nginx (F/OSS) and Nginx Plus, [click here](https://www.nginx.com/blog/whats-difference-nginx-foss-nginx-plus/).
+
+Structures for Nginx Plus have been built based on history of
+[status module documentation](http://nginx.org/en/docs/http/ngx_http_status_module.html)
+
+### Configuration:
+
+```
+# Read Nginx Plus' advanced status information
+[[inputs.nginx_plus]]
+  ## An array of Nginx status URIs to gather stats.
+  urls = ["http://localhost/status"]
+```
+
+### Measurements & Fields:
+
+- nginx_plus_processes
+  - respawned
+- nginx_plus_connections
+  - accepted
+  - dropped
+  - active
+  - idle
+- nginx_plus_ssl
+  - handshakes
+  - handshakes_failed
+  - session_reuses
+- nginx_plus_requests
+  - total
+  - current
+- nginx_plus_upstream, nginx_plus_stream_upstream
+  - keepalive
+  - zombies
+- nginx_plus_upstream_peer, nginx_plus_stream_upstream_peer
+  - requests
+  - unavail
+  - healthchecks_checks
+  - header_time
+  - response_time
+  - state
+  - active
+  - downstart
+  - healthchecks_last_passed
+  - weight
+  - responses_1xx
+  - responses_2xx
+  - responses_3xx
+  - responses_4xx
+  - responses_5xx
+  - received
+  - selected
+  - healthchecks_fails
+  - healthchecks_unhealthy
+  - backup
+  - responses_total
+  - sent
+  - fails
+  - downtime
+
+
+### Tags:
+
+- nginx_plus_processes, nginx_plus_connections, nginx_plus_ssl, nginx_plus_requests
+  - server
+  - port
+  - host
+
+- nginx_plus_upstream, nginx_plus_stream_upstream
+  - host
+  - upstream
+  - server
+  - port
+
+- nginx_plus_upstream_peer, nginx_plus_stream_upstream_peer
+  - id
+  - host
+  - upstream
+  - server
+  - port
+  - serverAddress
+
+### Example Output:
+
+Using this configuration:
+```
+[[inputs.nginx_plus]]
+  ## An array of Nginx Plus status URIs to gather stats.
+  urls = ["http://localhost/status"]
+```
+
+When run with:
+```
+./telegraf -config telegraf.conf -input-filter nginx_plus -test
+```
+
+It produces:
+```
+* Plugin: inputs.nginx_plus, Collection 1
+> nginx_plus_processes,server=localhost,port=12021,host=word.local respawned="0xc420075538" 1504922954000000000
```

Comment:
respawned is an address, this is probably a bug where we need to dereference a pointer.

Reply:
@danielnelson I think I fixed all of these. It looks like in the original commits the test was written like this, which was making the tests pass:

```
                       "selected": func() *int64 {
                               var v int64 = 1451606400000
                               return &v
                       }(),
```

I updated the test to just expect the int64:
```
                       "selected":               int64(1451606400000),
```

And updated the map:
```
"selected":               *peer.Selected,
```

Those were the only two instances where this was happening. I made the changes to both nginx_plus.go and the tests, and made sure this compiled. I then ran telegraf in test mode against one of our load balancers and I did not see any addresses in the output.

If this looks like a good fix I will update the README to remove the addresses from the sample output.

Reply:
You will want to check that they are not nil before dereferencing, I assume they are pointers because in some versions they don't exist.

Reply:
How would you recommend I do that? Something like this for respawned? ```
func (s *Status) gatherProcessesMetrics(tags map[string]string, acc telegraf.Accumulator) {
    var respawned int

    if s.Processes.Respawned != nil {
      respawned = *s.Processes.Respawned
    }

    acc.AddFields(
        "nginx_plus_processes",
        map[string]interface{}{
            "respawned": respawned,
        },
        tags,
    )

}
```

Reply:
Yeah

Reply:
Pushed a fix in https://github.com/influxdata/telegraf/pull/3214/commits/c84861aed74f64d42f6d4a7df71e7e376dd985d7